assets/img/2025-04-28-sparse-autodiff/demo/raw_code.jl

```diff
@@ -0,0 +1,56 @@
+import DifferentiationInterface as DI
+import SparseConnectivityTracer as SCT
+import SparseMatrixColorings as SMC
+import ForwardDiff as FD
+
+function differences(x)
+    n = length(x)
+    y = similar(x, n - 1)
+    for i in 1:(n - 1)
+        y[i] = x[i + 1] - x[i]
+    end
+    return y
+end
```

Comment:
Once again, I'm not sure what is gained by the namespaceing and custom implementation of `diff`.